src/pages/supplement/supplement.tsx

```diff
@@ -23,23 +46,58 @@ function Supplement() {
 
   return (
     <div className="d-flex m-5 justify-content-evenly">
-      <img src={query.data?.data.photo_url} className="col-5" alt="..." />
+      <img src={supplement?.photo_url} className="col-5" alt="..." />
```

Comment:
Because you're handling both the loading and error cases above, you should be able to access `query.data` (after the loading/error cases) and Typescript should understand that it's not `undefined` which would let you omit the `?` operators in most of your JSX.

IIUC the main reason you're assigning `const supplement = query.data?.data` is so you can pass values into your add to cart hook. You could instead pass those values into the `addItem` mutate function and avoid the early possible undefined assignment to `supplement`.